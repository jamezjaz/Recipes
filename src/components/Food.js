import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import React from 'react';
import FoodStyles from '../styles/FoodStyles.module.css';

const Food = props => {
  const { food } = props;
  return (
    <tbody>
      <tr>
        <td>
          <div className={FoodStyles.foodDetails}>
            <div>
              <img src={food.strMealThumb} className={FoodStyles.img} alt="foodImg" />
              <h4>{food.strMeal}</h4>
              <Link className={FoodStyles.link} to="/foodDetails">
                <span>Food Details</span>
              </Link>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  );
};

Food.propTypes = {
  food: PropTypes.shape({
    strMealThumb: PropTypes.string,
    strMeal: PropTypes.string,
  }).isRequired,
};

export default Food;
