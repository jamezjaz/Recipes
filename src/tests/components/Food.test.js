import React from 'react';
// import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Food from '../../components/Food';

// afterEach(cleanup);

const food = {
  foods: {
    meals: [{name: 'Chicken', category: 'Desert'},{name: 'Pizza', category: 'Starter'}]
  }
}

it('it renders correctly', () => {
  const tree = renderer
  .create(
    <Router>
        <Food food={food} />
    </Router>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});