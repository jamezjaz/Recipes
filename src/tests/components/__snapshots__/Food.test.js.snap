// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Food /> it renders correctly 1`] = `
<tbody>
  <tr>
    <td>
      <div
        className="foodDetails"
      >
        <div
          className="imgContainer"
        >
          <img
            alt="foodImg"
            className="img"
            data-testid="image"
          />
        </div>
        <h4
          data-testid="name"
        />
        <button
          className="btn btn-secondary"
          type="button"
        >
          <a
            className="link"
            href="/foodDetails/undefined"
            onClick={[Function]}
          >
            <h4>
              Food Details
            </h4>
          </a>
        </button>
      </div>
    </td>
  </tr>
</tbody>
`;
